### PR TITLE
Add trie node unit tests

### PR DESCRIPTION
**Pull request by blukat29**
## Proposed changes

- Add unit tests
  - storage/statedb/node_test.go tests `decodeNode()`
  - storage/statedb/hasher_test.go tests `hasher.hash()`
- Minor syntax improvement
  - `append(valueNode{}, val...)` -> `valueNode(val)`
- Silence print_trie_test.go

## Types of changes

- [ ] Bugfix
- [x] New feature or enhancement
- [ ] Others

## Checklist

- [x] I have read the [CONTRIBUTING GUIDELINES](https://github.com/klaytn/klaytn/blob/master/CONTRIBUTING.md) doc
- [x] I have signed the [CLA](https://cla-assistant.io/klaytn/klaytn)
- [x] Lint and unit tests pass locally with my changes (`$ make test`)
- [x] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

### CODE REVIEW

File: storage/statedb/hasher_test.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+func checkHasherHash(t *testing.T, idx int, tc *testNodeEncodingTC) {
+	hash := common.BytesToHash(tc.hash)
+
+	memDB := database.NewMemoryDBManager()
+	db := NewDatabase(memDB)
+	h := newHasher(nil)
+	defer returnHasherToPool(h)
+
+	hashed, cached := h.hash(tc.expanded, db, false)
```
Comment (jeongkyun-oh) [inline storage/statedb/hasher_test.go:19]: How about adding test cases for `hashRoot`?

File: storage/statedb/hasher_test.go [hunk 2]
```diff
@@ -0,0 +1,41 @@
+package statedb
```
Comment (jeongkyun-oh) [inline storage/statedb/hasher_test.go:1]: Missing copyright here

File: storage/statedb/node_test.go [hunk 3]
```diff
@@ -0,0 +1,188 @@
+package statedb
```
Comment (jeongkyun-oh) [inline storage/statedb/node_test.go:1]: Missing copyright here

**Review by jeongkyun-oh — APPROVED** [review]
LGTM except for minors

File: storage/statedb/hasher_test.go (hunk 1 above)
Comment (blukat29) [reply storage/statedb/hasher_test.go]: Added in [a87f5d8](https://github.com/klaytn/klaytn/pull/1843/commits/a87f5d8d4a1f4b0445674bb253855fcadf00f636).

File: storage/statedb/hasher_test.go (hunk 2 above)
Comment (blukat29) [reply storage/statedb/hasher_test.go:17]: Fixed in 5af4dd6.

File: storage/statedb/node_test.go (hunk 3 above)
Comment (blukat29) [reply storage/statedb/node_test.go:17]: Fixed in 5af4dd6.

**Review by jeongkyun-oh — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Review by aidan-kwon — APPROVED** [review]

**Pull request merged**